### PR TITLE
Fixed cooperation of Compound kernel and GMatP type

### PR DESCRIPTION
**Pull request by AsyaPronina**
<cut/>

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2020.2.0:16.04
build_image:Custom Win=openvino-2020.2.0
build_image:Custom Mac=openvino-2020.3.0

test_modules:Custom=gapi
test_modules:Custom Win=gapi
test_modules:Custom Mac=gapi

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Comment by alalek** [comment]
Is there some simple test for this case?

**Comment by dmatveev** [comment]
+1, please add a simple test on this @AsyaPronina

**Comment by asmorkalov** [comment]
@AsyaPronina could you add test to the patch?

**Comment by asmorkalov** [comment]
@AsyaPronina @rgarnov Friendly reminder.

**Comment by dmatveev** [comment]
@AsyaPronina ??

**Comment by AsyaPronina** [comment]
Hi @dmatveev , @asmorkalov ! Sorry for such late reply. Will add test soon.

**Comment by asmorkalov** [comment]
@AsyaPronina Friendly reminder.

**Comment by asmorkalov** [comment]
@AsyaPronina Friendly reminder.

File: modules/gapi/test/common/gapi_compoundkernel_tests.cpp [hunk 1]
```diff
@@ -227,6 +227,72 @@ namespace
...
+        static void run(const cv::Mat& in, cv::Mat& out)
+        {
+            constexpr int numPlanes = 3;
+            int inPlaneHeight = in.rows / numPlanes;
+
+            std::vector<cv::Mat> inPlanes;
+            for (int i = 0; i < numPlanes; ++i)
+            {
+                inPlanes.emplace_back(inPlaneHeight, in.cols, CV_MAKETYPE(in.depth(), 1),
+                                      in.data + i * (inPlaneHeight * in.step[0]));
```
Comment (dmatveev) [inline modules/gapi/test/common/gapi_compoundkernel_tests.cpp:261]: Better to use `cv::Mat::rowRange()`

**Review by dmatveev — APPROVED** [review]
"Approved with comments"

File: modules/gapi/test/common/gapi_compoundkernel_tests.cpp (hunk 1 above)
Comment (AsyaPronina) [reply modules/gapi/test/common/gapi_compoundkernel_tests.cpp]: Thanks a lot! Fixed!

**Pull request merged**